sdk/identity/Azure.Identity/src/ImdsManagedIdentityProbeSource.cs

```diff
@@ -169,10 +169,11 @@ protected override async ValueTask<AccessToken> HandleResponseAsync(bool async,
             return token;
         }
 
-        public static bool IsProbeRequest(HttpMessage message)
+        public static bool IsRetriableProbeRequest(HttpMessage message)
             => message.Request.Uri.Host == s_imdsEndpoint.Host &&
                 message.Request.Uri.Path == s_imdsEndpoint.AbsolutePath &&
-                !message.Request.Headers.TryGetValue(metadataHeaderName, out _);
+                !message.Request.Headers.TryGetValue(metadataHeaderName, out _) &&
+                !(message.Response.Content?.ToString().Contains("Identity not found") ?? false);
```

Comment:
Should we be specifying invariant culture for the contains check?

Reply:
good call out.